### ISSUE TITLE
add bottom margin to backend layout

### ISSUE THREAD

**Issue by JanMalte** [wagtail/wagtail#9199]
### Is your proposal related to a problem?

Using the backend on mobile devices it is hard to reach the buttons at the end of a page.

### Describe the solution you'd like

Just add some margin to the bottom so the buttons do not stick to the bottom edge of the screen.
This applies to all backend views, so this should be added in the basic layout file.

### Describe alternatives you've considered

None

### Additional context

Using safari on iPhone

**Comment by Scotchester** [comment]
@JanMalte Thanks for reporting. Could you add a screenshot illustrating the issue?

**Comment by JanMalte** [comment]
![Wagtail - Artikel bearbeiten](https://user-images.githubusercontent.com/544986/190210346-f87bbb16-c65c-4a14-805b-553a66e0bb68.png)

**Comment by lb-** [comment]
Calling this out as definitely a frustration, it is a small thing but it feels very 'constrained' at the bottom of the editing interface even on desktop.

<img width="1876" alt="Screen Shot 2022-09-29 at 7 15 18 am" src="https://user-images.githubusercontent.com/1396140/192890362-b95fc69d-55aa-46a9-88b2-8775a16d8ea0.png">
<img width="1874" alt="Screen Shot 2022-09-29 at 7 14 41 am" src="https://user-images.githubusercontent.com/1396140/192890374-25ef0252-4691-44d7-ac14-694ee21ae6a0.png">

Adding good first issue to this as it should be a small fix of a few rem at the bottom.

See also this potentially related issue https://github.com/wagtail/wagtail/issues/8664

**Comment by lb-** [comment]
@JanMalte  I have re-worded your issue - please check I have understood the problem correctly. I have also renamed to avoid the 'backend' term as this can be a bit confusing, we normally call this just the 'admin'. Finally, I have added a potential solution with a screenshot.

**Comment by Ruth-ikegah** [comment]
Hi @lb- and @JanMalte I am an outreach potential intern interested in the I would love to fix this issue as my first contribution to the Wagtail project. Please could you assign this issue to me?

**Comment by mohammadareeb95** [comment]
On adding `padding-bottom:30px` this is what we get. I'm not sure about the purple vertical bar that's hanging there.
![image](https://user-images.githubusercontent.com/77102111/194768806-6c915d88-a4cc-4575-aa34-fd93be6dd5e8.png)

**Comment by mohammadareeb95** [comment]
### Adding `padding-bottom: 10rem` gives

![image](https://user-images.githubusercontent.com/77102111/194768955-b4d4d699-6cf9-4110-9ac0-c05c3050183b.png)

**Comment by Scotchester** [comment]
I think the original issue has been misinterpreted. As I understand it, it is related to the scroll behavior of iOS.

If I open the bakerydemo in iOS Safari, go to edit any page, and fling the page to reach the bottom in one scrolling motion, the bottom action menu pops into view briefly and then bounced below the toolbar, resulting in this:

![screenshot of edit screen on iOS not showing action menu](https://user-images.githubusercontent.com/1044670/194774064-1fa1a3a7-e4bf-4274-bf6c-0ae7e97a48dd.png)

You have to then scroll the page again in order to get the menu in view and able to be interacted with.

**Comment by nadaHan** [comment]
Please Assign to me  @JanMalte  <3

**Comment by Lovelyfin00** [comment]
Hi
I'm an outreachy applicant and would love to work on this issue @Scotchester

**Comment by damli40** [comment]
would love to work on this

**Comment by thibaudcolas** [comment]
Hi folks, no need for you to ask – just show us you understand the problem, explain what you intend to do, and you can get going with making a pull request. It’s better if only one person works on a given problem at a time, so this is normally first-come-first-served.

**Comment by mohammadareeb95** [comment]
> I think the original issue has been misinterpreted. As I understand it, it is related to the scroll behavior of iOS.
> 
> If I open the bakerydemo in iOS Safari, go to edit any page, and fling the page to reach the bottom in one scrolling motion, the bottom action menu pops into view briefly and then bounces below the toolbar, resulting in this:
> 
> You have to then scroll the page again in order to get the menu in view and able to be interacted with.

@Scotchester so is there no need to add space in the footer?

 Do you think it's only a problem on mobile devices?

As in the screenshots above, adding `padding-bottom` is improving the UI. 
If we agree to move in this direction please let me know what would be a better approach:

1.  Add ` padding-bottom: 30 px` 
2.  Add `padding-bottom: 10rem`

**Comment by PreciousOritsedere** [comment]
@mohammadareeb95 Hi, are you still working on this issue?

**Comment by Scotchester** [comment]
@mohammadareeb95 After re-reading the whole thread (including the original post @JanMalte made before @lb-'s edit), I'm still not entirely sure what exact issue Jan Malte was reporting. I am still not sure I agree with LB's interpretation. It's possible my interpretation is also incorrect, but is still a problem (and may warrant breaking out into a separate issue).

@JanMalte Could you please try to clarify what you originally meant when raising this issue?

@mohammadareeb95 Whatever Jan Malte's answer will be, I disagree that the proposed change shown in your screenshots is an improvement. The design calls for the buttons to stick to the bottom of the window, and I don't think adding padding to push them up is solving an actual problem. I also suspect it would create a problem if a page editing form is long enough that fields would go below the buttons, appearing between that and the bottom of the window. That would be very confusing, visually. Let's get clarification on the original intent and take it from there.

**Comment by lb-** [comment]
Sorry if my interpretation was incorrect, nonetheless I still feel there is not enough 'breathing' room bellow the last form controls and the footer button actions. Happy to split this concern out and revert this issue to the original problem but I understand that adding more margin should solve both the iOS specific and general browser issue. 

There is also the potential usage of the mobile viewport height specific units. 

See https://css-tricks.com/the-trick-to-viewport-units-on-mobile/

**Comment by Scotchester** [comment]
> nonetheless I still feel there is not enough 'breathing' room bellow the last form controls and the footer button actions

Yes! Meant to say that I agree with that, too. I'm just not 100% sure that's what the original issue was about, and also I don't think it will address the concern I noted with how the sticky footer gets stuck below the iOS Safari toolbar on first scrolling to the bottom, because the sticky footer will still be in the same relative position.

**Comment by lb-** [comment]
Here is the original content - click 'edited by lb' and the small dropdown to see any history of any issue or discussion description.

<img width="522" alt="Screen Shot 2022-10-13 at 3 56 19 pm" src="https://user-images.githubusercontent.com/1396140/195513779-21deb8e3-7d6a-4ad5-98fd-e628bb244624.png">

<img width="633" alt="Screen Shot 2022-10-13 at 3 57 31 pm" src="https://user-images.githubusercontent.com/1396140/195513900-6106903f-5bbf-475c-8481-943fc88201c6.png">

> raw text of original description below

Is your proposal related to a problem?

Using the backend on mobile devices it is hard to reach the buttons at the end of a page.
Describe the solution you'd like

Just add some margin to the bottom so the buttons do not stick to the bottom edge of the screen.
This applies to all backend views, so this should be added in the basic layout file.
Describe alternatives you've considered

None
Additional context

Using safari on iPhone

**Comment by mohammadareeb95** [comment]
> @mohammadareeb95 Hi, are you still working on this issue?

@PreciousOritsedere yes!

**Comment by mohammadareeb95** [comment]
Some points about this issue: 
### When on desktop, the `Save draft` button sticks to the screen. Meaning you can see it, without going to the very bottom
![image](https://user-images.githubusercontent.com/77102111/197508580-fac56a59-e0b4-457e-b713-b9413f091388.png)



### When on mobile device, one has to scroll the entire page to get to the `Save draft` button.
![image](https://user-images.githubusercontent.com/77102111/197508409-9c92c98c-e292-4ce8-9e57-0cdcda382ee6.png)

**Comment by mohammadareeb95** [comment]
Some questions: 

- Do we need a sticky button to the interface in both views (desktop/mobile)?
- Should we just go around giving some breathing room for the `Save draft` button

**Comment by mohammadareeb95** [comment]
@lb- @thibaudcolas please let me know how move forward in this issue

**Comment by JanMalte** [comment]
>     * Do we need a sticky button to the interface in both views (desktop/mobile)?
I prefer the button at the bottom of the page on mobile screens, as the space is very limited and a sticky button reduces the available space even more.

>     * Should we just go around giving some breathing room for the `Save draft` button
On mobile screens some margin from the bottom would be great, from my point of view.

**Comment by mohammadareeb95** [comment]
I agree

**Comment by Scotchester** [comment]
> > Do we need a sticky button to the interface in both views (desktop/mobile)?
> 
> I prefer the button at the bottom of the page on mobile screens, as the space is very limited and a sticky button reduces the available space even more.

I agree with this, as well. Plus, it's "easier" on a touch device to quickly scroll to the bottom of the page.

> > Should we just go around giving some breathing room for the `Save draft` button
> 
> On mobile screens some margin from the bottom would be great, from my point of view.

I'm _still_ not sure what you originally meant, or what exactly this latest statement means, @JanMalte. Margin between the last field in the edit form and the save/publish button (as LB posited)? Margin between the save/publish button and the ultimate bottom of the page (and thus the browser chrome)? Something else entirely?

Could you mock up exactly what you mean?

**Comment by Scotchester** [comment]
Thanks for clarifying!

Now I want to push a little further to make sure that is the right solution :)

You originally said that the save/publish button is "hard to reach". Can you explain why that is?

Could there be any alternate solutions? I notice that the button is pretty short in height, so perhaps just making it taller would improve the ability to hit it accurately?

**Comment by JanMalte** [comment]
> You originally said that the save/publish button is "hard to reach". Can you explain why that is?

If you make one fast swipe gesture to scroll to the bottom of the page, the iOS safari browser bumps back from the bottom a little bit. Not sure this is on all pages or just the wagtail admin due to some CSS of the page or button.
You have to scroll up little and slowly scroll down to get the button visible/reachable.

**Comment by Scotchester** [comment]
> > You originally said that the save/publish button is "hard to reach". Can you explain why that is?
> 
> If you make one fast swipe gesture to scroll to the bottom of the page, the iOS safari browser bumps back from the bottom a little bit. Not sure this is on all pages or just the wagtail admin due to some CSS of the page or button. You have to scroll up little and slowly scroll down to get the button visible/reachable.

Ahah! So it **was** what I originally thought 😁 

OK then, this needs a lot more testing and experimentation (with multiple browsers, not just Safari) to figure out the right approach to this.

@mohammadareeb95, do you have an iOS device, or a way to emulate it?

**Comment by mohammadareeb95** [comment]
@Scotchester Yes, I do have an iphone but not a Macbook. (I use Linux instead)

**Comment by Scotchester** [comment]
> @Scotchester Yes, I do have an iphone but not a Macbook. (I use Linux instead)

Would you be able to use your iPhone to do some testing of the current behavior (particularly with regards to what happens to the button if you scroll very quickly to the bottom) on Safari, Chrome, and Firefox, and report back with your findings?

**Comment by mohammadareeb95** [comment]
sure!

**Comment by mohammadareeb95** [comment]
## Safari

https://user-images.githubusercontent.com/77102111/197854555-e1f1938f-da7f-4ca3-b1e3-7cf76d9ff539.mp4

## Chrome


https://user-images.githubusercontent.com/77102111/197859471-9892f803-ce95-4ea8-8eef-35c2d8031b8a.mp4

## Firefox

https://user-images.githubusercontent.com/77102111/197859574-d174eca7-a7ee-48a4-949b-405a22c101d3.mp4



- I can confirm that very quick scroll in Safari bumps back and hides the save draft button. But, I also noticed this behaviour on Tab refresh. Meaning if I try to scroll very quickly another time, it doesn't bump back. On the contrary, if I refresh or load the page again, bumping can be visible.
- This behaviour is not found in Chrome or Firefox

**Comment by Scotchester** [comment]
Thanks for the videos, @mohammadareeb95! Fascinating stuff. I don't think that merely adding margin under the button is the best solution for this, as it will negatively impact the experience of everyone except for Mobile Safari users (and even Mobile Safari users, as they would then be able to scroll the page "too far"). I guess the next step is to do some research and see if anyone else has run into a similar issue with Mobile Safari (with any site/app, not just Wagtail) and if/how they were able to fix it.

**Comment by Redidacove** [comment]
plz assign this issue to me sir asap

**Comment by lb-** [comment]
@Redidacove feel free to give this a go - but read through the previous comments and linked issues and PRs in full. There is a bit of nuance to this. 

It actually may not be suitable as a good first issue, but see how you go. We don't normally assign issues outside of the core team. 

Thanks.

**Comment by Redidacove** [comment]
Thnks I will try still if I could then will raise a pr

**Comment by syedimtiyazali20** [comment]
I want to work on this issue. Please, assign me this issue.

**Comment by lb-** [comment]
@SyedImtiyaz-1 I recommend you always review the comments on the issue before trying to pick it up. See above, just yesterday, Redidacove just said they would be working on this.

**Comment by lb-** [comment]
Removing good first issue label as this solution is likely small but needs a good level of understanding of both CSS and all potential layouts of all editing interfaces in Wagtail. Plus needs to be tested on all browsers extensively as it's been found to cause different problems in different browsers.

**Comment by shindodkar** [comment]
@lb-  is this close?

**Comment by lb-** [comment]
@shindodkar I don't think anyone has come close to putting up a solid PR that solves this issue yet sorry.

**Comment by lb-** [comment]
@soh55 - I noticed you are modifying the 'changes summary' content in the footer.

My I suggest you explore using a `HelpPanel` for this purpose, it's more suited to showing content inside the editing interface. https://docs.wagtail.org/en/stable/reference/pages/panels.html#helppanel

You could use some CSS to make this sticky if needed.

Thanks for the suggestion of the pseudo element CSS, I am not sure it will fix for Firefox/Safari but if anyone wants to pick this issue up they can look into this approach.

**Comment by thibaudcolas** [comment]
Closing, as it seems to me we resolved it when redesigning the spacing between sections of the editor. Happy to reconsider if there are scenarios that are still problematic
